backend/src/scripts/utility/teamspaces/removeIncompleteRevisions.js

```diff
@@ -0,0 +1,214 @@
+			const fileRefs = record[refAttribute];
+			if (fileRefs) {
+				// handle different ref formats
+				// record refs currently stored in the following formats:
+				// 1) { ref: 'refString' }
+				// 2) { ref: {
+				//        key1: 'key1RefString',
+				//        key2: 'key2RefString',
+				//    }
+				if (TypeChecker.isString(fileRefs)) {
+					filenames.push(fileRefs);
+				} else if (TypeChecker.isObject(fileRefs)) {
+					// for record removal, we want to remove all refs,
+					// i.e. 'key1RefString' and 'key2RefString',
+					// do not need to discern between keys
+					for (const entry of Object.values(fileRefs)) {
+						filenames.push(entry);
+					}
+				} else {
+					logger.logError(`Unsupported record type: ${Object.prototype.toString.call(fileRefs)}`);
+				}
+			}
+
+			return fileRefs;
+		});
+
+		await processFilesAndRefs(teamspace, collection, { _id: { $in: filenames } });
+	}
+
+	await deleteMany(teamspace, collection, filter);
+};
+
+const processModelStash = async (teamspace, model, revId) => {
+	const modelStashPromises = [];
+
+	modelStashPromises.push(processFilesAndRefs(
+		teamspace,
+		`${model}.stash.json_mpc`,
+		{ filename: { $regex: `^/${teamspace}/${model}/revision/${UUIDToString(revId)}/` } },
+	));
+
+	modelStashPromises.push(processFilesAndRefs(
+		teamspace,
+		`${model}.stash.json_mpc.ref`,
+		{ _id: { $regex: `^${UUIDToString(revId)}/` } },
+	));
+
+	const supermeshIds = (await find(teamspace, `${model}.stash.3drepo`, { rev_id: revId, type: 'mesh' }, { _id: 1 })).map((r) => UUIDToString(r._id));
+	supermeshIds.map((supermeshId) => {
+		modelStashPromises.push(processFilesAndRefs(teamspace, `${model}.stash.3drepo.ref`, { _id: { $in: [
+			`${supermeshId}_faces`,
+			`${supermeshId}_normals`,
+			`${supermeshId}_vertices`,
+		] } }));
+		modelStashPromises.push(processFilesAndRefs(teamspace, `${model}.stash.json_mpc.ref`, { _id: { $in: [
+			`${supermeshId}.json.mpc`,
+			`${supermeshId}_unity.json.mpc`,
+		] } }));
+		modelStashPromises.push(processFilesAndRefs(teamspace, `${model}.stash.src.ref`, { _id: { $in: [
+			`${supermeshId}.src.mpc`,
+		] } }));
+		modelStashPromises.push(processFilesAndRefs(teamspace, `${model}.stash.unity3d.ref`, { _id: { $in: [
+			`${supermeshId}.unity3d`,
+		] } }));
+
+		return supermeshId;
+	});
+
+	modelStashPromises.push(removeRecords(teamspace, `${model}.stash.unity3d`, { _id: revId }));
+	modelStashPromises.push(removeRecords(teamspace, `${model}.stash.3drepo`, { rev_id: revId }, '_extRef'));
+
+	await Promise.all(modelStashPromises);
+};
+
+const processModelSequences = async (teamspace, model, revId) => {
+	const sequences = await find(teamspace, `${model}.sequences`, { rev_id: revId }, { frames: 1 });
+	const sequenceFilenames = [];
+
+	for (const { _id, frames } of sequences) {
+		if (frames) {
+			for (const { state } of frames) {
+				sequenceFilenames.push(state);
+			}
+		}
+
+		// eslint-disable-next-line no-await-in-loop
+		await Promise.all([
```

Comment:
I don't think we need to delete this one sequence at a time 

Just collect all the info and run it in a single query